proposals/0448-type-variable-scoping.rst

```diff
@@ -198,42 +233,74 @@ All other variables are explicitly bound, and the inconsistency means more to le
 Also, implicit syntax in general allows the beginner to not realize what they are doing.
 What are tedious tasks for the expert may be helpful learning steps to them.
 
-Further, the most beginnning students may be taught with both ``-XNoImplicitForAll`` and ``-XNoExplicitForAll``.
+Further, most beginning students may be taught with both ``-XImplicitBinds``, ``-XNoExplicitForAll``, and ``-XNoPolyKinds``.
 This means it's impossible to write forall types by any means.
 Combine with ``-Wmissing-signatures`` and ``-Wmissing-local-signatures``, so inferred polymorphic types of bindings are also prohibited, and a monomorphic custom prelude, and forall types are all but expunged entirely.
 
-I don't wish to argue whether these choices do or don't actually help learning, but just state that some people have opinions that they do and there is no technical reason GHC cannot accommodate them.
+@Ericson2314 doesn't wish to argue whether these choices do or don't actually help learning, but just state that some people have opinions that they do and there is no technical reason GHC cannot accommodate them.
+
+Consistency
+^^^^^^^^^^^
+
+Notice how today that out-of-scope variables in negative position signatures are implicitly bound in *different* ways depending on whether they are type variables (in pattern signatures) or kind variables (in negative position kind signatures).
+By banning implicit binding, we side-step that difference.
+
+After all, given::
+
+  data Foo (a :: k)
+
+desugars to::
+
+  data Foo @k (a :: k)
+
+a new Haskeller might conceivably think::
+
+  \(Foo (a :: k) -> ..
+
+desugars to::
+
+  \(Foo @k (a :: k) ->
+
+or::
+
+  \ @k (Foo (a :: k) ->
+
+which happen to be true in some simple common cases, but are in fact incorrect in general.
+
+That it takes a complicated example to show why these false desugarings aren't true in general make this is a huge educational stumbling block!
 
 Unified Namespace
 ^^^^^^^^^^^^^^^^^
 
 If `#270`_ is accepted, there will be a way to program Haskell with "morally" one namespace for types and terms alike.
-However, there is one exception to the unification of namespaces: lower case variables in type signatures bound "like terms" still are treated as free and implicitly bound with a ``forall`` instead::
+However, there is one exception to the unification of namespaces: lower case variables in type signatures bound "like terms" still are treated as free and implicitly bound instead::
 
   t = Int
-  x :: t -- sugar for 'forall t. t', not 't ~ Int'
+  x :: t -- sugar for 'forall t. t', not a use of 't' resolving to 'Int'
```

Comment:
.

Reply:
Perhaps I was being too liberal modifying this because...it is in fact my own writing taking verbatim from #285 here that I am nit-picking!

The issue that is that `t` is *not* sugar for `t ~ Int`, because `t` is a type and `t ~ Int` is a constraint. I was in #285 trying to express that `t` is bound as `Int` but did so in what strikes me in hindsight as a sloppy manner.

I suppose I could do this in a separate PR or not at all. Let me know what you prefer.